HyperVExtension/src/HyperVExtension/Helpers/DevSetupAgentDeploymentHelper.cs

```diff
@@ -0,0 +1,306 @@
+
+                # Stop and remove previous version of DevSetupAgent service if it exists
+                $service = Get-Service -Name $using:DevSetupAgentConst -ErrorAction SilentlyContinue
+                if ($service -ne $null)
+                {
+                    $serviceWMI = Get-WmiObject -Class Win32_Service -Filter ""Name='$using:DevSetupAgentConst'""
+                    $existingServicePath = $serviceWMI.Properties[""PathName""].Value
+                    if ($existingServicePath)
+                    {
+                        $guestDevSetupAgentPath = Split-Path $existingServicePath -Parent
+                    }
+                    
+                    try
+                    {
+                        Write-Host ""Stopping DevSetupAgent service""
+                        Write-Progress -Activity $using:activity -Status ""Stopping DevSetupAgent service $destinationPath"" -PercentComplete 30
+                        $service.Stop()
+                    }
+                    catch
+                    {
+                        Write-Host ""Ignoring error: $PSItem""
+                    }
+
+                    Remove-Variable -Name service -ErrorAction SilentlyContinue
+
+                    # Remove-Service is only available in PowerShell 6.0 and later. Windows doesn't come with it preinstalled.
+                    Write-Host ""Removing DevSetupAgent service""
+                    Write-Progress -Activity $using:activity -Status ""Removing DevSetupAgent service"" -PercentComplete 35
+                    $serviceWMI = Get-WmiObject -Class Win32_Service -Filter ""Name='$using:DevSetupAgentConst'""
+                    $serviceWMI.Delete()
+                    Remove-Variable -Name serviceWMI -ErrorAction SilentlyContinue
+                }
+
+                # Stop previous version of DevSetupEngine CON server if it exists
+                $devSetupEngineProcess = Get-Process -Name ""$using:DevSetupEngineConst"" -ErrorAction SilentlyContinue
+                if ($devSetupEngineProcess -ne $null)
+                {
+                    Write-Host ""Stopping $using:DevSetupEngineConst process""
+                    Write-Progress -Activity $using:activity -Status ""Stopping $using:DevSetupEngineConst process"" -PercentComplete 40
+                    Stop-Process -Force -Name ""$using:DevSetupEngineConst""
+                }
+            
+                # Remove previous version of DevSetupAgent service files
+                if (Test-Path -Path $guestDevSetupAgentPath)
+                {
+                    Write-Host ""Deleting old DevSetupAgent service files""
+                    Write-Progress -Activity $using:activity -Status ""Deleting old DevSetupAgent service files"" -PercentComplete 45
+                    Remove-Item -Recurse -Force -Path $guestDevSetupAgentPath
+                }
+
+                if ($using:isDirectory)

```

Comment:
interesting, do we need the using statement here if these are just bools?

Reply:
Yes, it's capturing variable from the part of the script that runs on host to be able to access it from script that runs on guest.